src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs

```diff
@@ -367,7 +367,8 @@ internal List<TargetSpecification> GetDependencies(ProjectLoggingContext project
                     projectLoggingContext.BuildEventContext);
                 _state = TargetEntryState.Completed;
 
-                if (!projectLoggingContext.LoggingService.OnlyLogCriticalEvents)
+                if (projectLoggingContext.LoggingService.MinimumRequiredMessageImportance > MessageImportance.Low &&
```

Comment:
When `Verbosity` is diagnostic, this becomes true:

https://github.com/dotnet/msbuild/blob/98fe1859f5bef8f4add8a31cc09e13ce08391a73/src/Build/Logging/ConsoleLogger.cs#L486-L489

then the added condition in this PR is false?

Conceptually, if `MinimumRequiredMessageImportance` is `Low`, we want to log **more**, not less? Or am I missing something here?

My understanding of MinimumRequiredMessageImportance is, "user want low importance message and above". So, the lower the value, the more we should log?

@rainersigwald @YuliiaKovalova

Reply:
I cannot believe that enum is ordered backward. Will fix.